Python/Product/Debugger/Debugger/LegacyDebuggerProtocol.cs

```diff
@@ -0,0 +1,593 @@
+            public string condition;
+        }
+
+        public sealed class SetBreakpointPassCountRequest : GenericRequest {
+            public const string Command = "legacySetBreakpointPassCount";
+
+            public override string command => Command;
+
+            public int breakpointId;
+            public BreakpointPassCountKind passCountKind;
+            public int passCount;
+        }
+
+        public sealed class SetBreakpointHitCountRequest : GenericRequest {
+            public const string Command = "legacySetBreakpointHitCount";
+
+            public override string command => Command;
+
+            public int breakpointId;
+            public int hitCount;
+        }
+
+        public sealed class GetBreakpointHitCountRequest : Request<GetBreakpointHitCountResponse> {
+            public const string Command = "legacyGetBreakpointHitCount";
+
+            public override string command => Command;
+
+            public int breakpointId;
+        }
+
+        public sealed class GetBreakpointHitCountResponse : Response {
+            public int hitCount;
+        }
+
+        public sealed class ResumeAllRequest : GenericRequest {
+            public const string Command = "legacyResumeAll";
+
+            public override string command => Command;
+        }
+
+        public sealed class ResumeThreadRequest : GenericRequest {
+            public const string Command = "legacyResumeThread";
+
+            public override string command => Command;
+
+            public long threadId;
+        }
+
+        public sealed class AutoResumeThreadRequest : GenericRequest {
+            public const string Command = "legacyAutoResumeThread";
+
+            public override string command => Command;
+
+            public long threadId;
+        }
+
+        public sealed class ClearSteppingRequest : GenericRequest {
+            public const string Command = "legacyClearStepping";
+
+            public override string command => Command;
+
+            public long threadId;
+        }
+
+        public sealed class ExecuteTextRequest : GenericRequest {
+            public const string Command = "legacyExecuteText";
+
+            public override string command => Command;
+
+            public string text;
+            public int executionId;
+            public long threadId;
+            public int frameId;
+            public FrameKind frameKind;
+            public ReprKind reprKind;
+        }
+
+        public sealed class DetachRequest : GenericRequest {
+            public const string Command = "legacyDetach";
+
+            public override string command => Command;
+        }
+
+        public sealed class SetExceptionInfoRequest : GenericRequest {
+            public const string Command = "legacySetExceptionInfo";
+
+            public override string command => Command;
+
+            public int defaultBreakOnMode;
+            public ExceptionInfo[] breakOn;
```

Comment:
>breakOn [](start = 35, length = 7)

Although the intermediate representation in PythonProcess is in a Enumerable<KeyValuePair<string,int>, the original data from AD7Engine is stored in a dictionary<string,int>, and the final destination on Python side is also a dictionary.  It feels like it could be transported in json as a dictionary too.